### PR TITLE
Added support for private docker registry

### PR DESCRIPTION
**Pull request by bdpiparva**
* Supports private docker registry
* Renamed `ValidatePluginSettings` to `ValidatePluginSettingsRequest`
* Improved plugin-settings UI

### CODE REVIEW

File: src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java [hunk 1]
```diff
@@ -47,23 +47,83 @@ public void shouldValidateABadConfiguration() throws Exception {
...
+        settings.put("enable_private_registry_authentication", "true");
+        settings.put("private_registry_server", "");
+        settings.put("private_registry_username", "");
+        settings.put("private_registry_password", "");
+        settings.put("auto_register_timeout", "10");
+        GoPluginApiResponse response = new ValidateConfigurationExecutor(settings).execute();
+
+        String expectedString = "[\n" +
+                "  {\n" +
+                "    \"message\": \"private_registry_server must not be blank.\",\n" +
```
Comment (ketan) [inline src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java]: Did you mean to say `Private registry server must not be blank`?

File: src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java [hunk 2]
```diff
@@ -47,23 +47,83 @@ public void shouldValidateABadConfiguration() throws Exception {
...
+                "    \"key\": \"private_registry_server\"\n" +
+                "  },\n" +
+                "  {\n" +
+                "    \"message\": \"private_registry_username must not be blank.\",\n" +
```
Comment (ketan) [inline src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java]: Same as above?

File: src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java [hunk 3]
```diff
@@ -47,23 +47,83 @@ public void shouldValidateABadConfiguration() throws Exception {
...
+                "    \"key\": \"private_registry_username\"\n" +
+                "  },\n" +
+                "  {\n" +
+                "    \"message\": \"private_registry_password must not be blank.\",\n" +
```
Comment (ketan) [inline src/test/java/cd/go/contrib/elasticagents/dockerswarm/elasticagent/executors/ValidateConfigurationExecutorTest.java]: Same as above?

**Comment by bdpiparva** [comment]
Screenshots:

1)

screen shot 2017-05-10 at 11 45 05 am

2)
screen shot 2017-05-10 at 11 45 46 am

**Pull request merged**